### ISSUE TITLE
Suggestion: use FEN from PGN

### ISSUE THREAD

**Issue by jensguballa** [mliebelt/PgnViewerJS#137]
For alternative starting positions the FEN-tag from the PGN can be used, and thus there is no need to pass it as a separate configuration option to PgnView(). This probably is only applicable when viewing a PGN and does not apply when just the board position is provided.

Reference: http://www.saremba.de/chessgml/standards/pgn/pgn-complete.htm#c9.7.2

**Comment by mliebelt** [comment]
Please have a look at #129 

This should be already fixed, at least under the [Configuration Builder](http://mliebelt.github.io/PgnViewerJS/docu/example/config.html), you can try the PGN given in that issue. That works for me.

Perhaps you have not included `[SetUp "1"]` as expected in the documentation?

**Comment by jensguballa** [comment]
Although your guess was right, and my chess data base (Scid) did not include the "SetUp" tag in the PGN (I need to file a bug report for it), I am encountering still problems with PgnViewerJS.

I used the following PGN with the SetUp tag added:
```
[Event "Firmenschach SG Stern 1 - SEL 1"]
[Site ""]
[Date "1991.04.22"]
[Round "?"]
[White "Krämer"]
[Black "Guballa"]
[Result "0-1"]
[Setup "1"]
[FEN "5R2/1q6/8/7p/7P/6P1/4kP2/6K1 b - - 6 63"]

63...Qe4 ( 63...Qb1+ 64.Kg2 Qf1+ 65.Kh2 {Hm, und jetzt? Wenn die Dame mit Tempogewinn wegziehen könnte, dann käme der schwarze König nach f1. So aber:} 65...Qd1 66.Kg2 {und die Festung steht. Doch genau dieser Gedankengang brachte mich auf die Lösung! Was wäre, wenn der Turm nicht auf f8, sondern irgendwo anders auf der f-Linie stehen würde?} ) 64.Rf4 Qa8!! {Zugzwang! Der Turm muss das geschützte Feld f4 verlassen, kann aber nicht nach f8.} 65.Rf7 {Noch das hartnäckigste.} ( 65.Rf6? Qa1+ 66.Kg2 Qxf6 ) ( 65.Rf5 Qa1+ 66.Kg2 Qf1+ 67.Kh2 Qb1! {Mit Tempogewinn wird jetzt das Feld f1 geräumt.} 68.Rf8 Kf1 69.Rf4 Qb7 {Droht Dg2 matt.} 70.f3 Qb2+ 71.Kh3 Qg2# ) ( 65.Rb4 Qa1+ 66.Kg2 Qf1+ 67.Kh2 Qxf2+ 68.Kh3 Qg1 -+ {und Schwarz gewinnt, z.B.: } 69.Rb2+ Kd3 70.Rg2 Qh1+ 71.Rh2 Qf1+ 72.Rg2 Ke3 73.Kh2 Kf3 74.Rb2 Qd1 75.Ra2 Qc1 76.Rg2 Qe1 77.Rg1 Qf2+ 78.Kh1 Qe2 {Wieder Zugzwang.} 79.g4 hxg4 80.h5 g3 ) 65...Qa1+ 66.Kg2 Qf1+ 67.Kh2 Qb1 {Zeitverlust, aber wegen der Drohung Kf1 verdirbt dieser Zug nichts.} 68.Kg2 Qf1+ 69.Kh2 Qd1! {Jetzt aber...} 70.g4 {Verzweiflung, aber es gibt nichts besseres.} ( 70.Kg2 Qd5+ 71.Kg1 Qxf7 ) 70...Kf1 ( 70...Kf1 71.gxh5 Qg4 {Und Matt im nächsten Zug.} ) 0-1
```
If I am using the Configuration Builder only providing the PGN text area, it fails with "No legal move: Qe4".

If I now copy the FEN string from the PGN and provide it additionally into the FEN-text field, everything works as expected.

Using the PGN from https://github.com/mliebelt/PgnViewerJS/issues/129 does not cause the problem described above.

**Comment by mliebelt** [comment]
Ok, I will have a look at that. That may be anything ... Ok, tried again, and found the "bug": The reason is, that in the specification, the tag is named `SetUp`, but in your notation, you use `Setup` with a lowercase `u`. I will change that and allow all tags to have any cases.

Btw, I was long ago member of the chess club in Ostfildern, where the Krämer brothers played ... :-) Funny times.

**Comment by jensguballa** [comment]
Ups, you are right, I did not pay attention to the case of the tag. If I am using the correct case, everything is working fine. So I would not mind if you just close this issue, as I believe the tags are case sensitive and the implementation should not be adapted to dump users. ;-)

From chess point of view I am currently more active in the area north of Stuttgart ("Unterland"). Small world...

**Comment by jensguballa** [comment]
I just recognized: when I was reseaching for the SetUp tag, I came across this site: https://www.schachbundesliga.de/weisheiten-im-endspiel/weisheiten-im-endspiel-37. 

On this site a PGN with endgame positions can be downloaded, and within this PGN file the tag "Setup" is used ("lower case u", I probably copied it from there). 

Not sure, if this PGN is generated by ChessBase, but if I remember correctly, long time ago ChessBase used 0-0-0 (Zeros) instead of O-O-O ("capital letter o"). Anyway, being more flexible for parsing PGN can be beneficial.
